src/services/geometry/richgeo/IrtGeoDRICH.cc

```diff
@@ -186,7 +183,24 @@ void richgeo::IrtGeoDRICH::DD4hep_to_IRT() {
   // define the `cell ID -> pixel position` converter
   SetReadoutIDToPositionLambda();
 }
-
+TVector3 richgeo::IrtGeoDRICH::GetSensorSurfaceNorm(CellIDType id){
+  TVector3 sensorNorm;
+  auto cellMask = uint64_t(std::stoull(m_det->constant<std::string>("DRICH_cell_mask")));
+  auto sensor_info = this->m_sensor_info;
+  auto sID = id & cellMask;
+  auto sensor_info_it = sensor_info.find(sID);
+  if(sensor_info_it!=sensor_info.end()){
+    auto sensor_obj = sensor_info_it->second;
+    auto normZdir = sensor_obj.surface_offset.Unit();
+    sensorNorm.SetX(static_cast<double>(normZdir.x()));
+    sensorNorm.SetY(static_cast<double>(normZdir.y()));
+    sensorNorm.SetZ(static_cast<double>(normZdir.z()));
+  }
+  else
+    m_log->error("Cannot find sensor {} in IrtGeoDRICH::GetSensorSurface", id);
```

Comment:
Throw exception instead of just print?

Reply:
Hi @wdconinc, I can throw an exception. I was imagining the following:
`throw std::runtime_error("sensor not found in IrtGeoDRIC::GetSensorSurfaceNormal");`
But can we also print the `sensorID` that was not found in the exception? I can imagine to keep both lines inside the else condition.

Reply:
You will have to put the logging line before the throw.